nautobot/core/ui/object_detail.py

```diff
@@ -153,6 +166,98 @@ def get_extra_context(self, context: Context):
         return {}
 
 
+class Button(Component):
+    """Base class for UI framework definition of a single button within an Object Detail (Object Retrieve) page."""
+
+    def __init__(
+        self,
+        *,
+        label,
+        color=ButtonColorChoices.DEFAULT,
+        link_name=None,
+        icon=None,
+        template_path="components/button/default.html",
+        required_permissions=None,
+        javascript_template_path=None,
+        attributes=None,
+        **kwargs,
+    ):
+        """
+        Initialize a Button component.
+
+        Args:
+            label (str): The text of this button, not including any icon.
+            color (ButtonColorChoices): The color (class) of this button.
+            link_name (str, optional): View name to link to, for example "dcim:locationtype_retrieve".
+                This link will be reversed and will automatically include the current object's PK as a parameter to the
+                `reverse()` call when the button is rendered.
```

Comment:
Don't we need and option for "external links" as well, to just pass the `https://...` url?

Reply:
We undoubtedly will at some point, yeah.

Reply:
Added `get_link()` method for ease of extension by subclasses.